opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -304,8 +304,15 @@ namespace Opm
         bool isIORank = output_ ;
         if( parallelOutput_ && parallelOutput_->isParallel() )
         {
+            // If this is not the initial write and no substep, then the well
+            // state used in the computation is actually the one of the last
+            // step. We need that well state for the gathering. Otherwise
+            // It an exception with a message like "global state does not
+            // contain well ..." might be thrown.
+            int wellStateStepNumber = ( ! substep && timer.reportStepNum() > 0) ?
+                (timer.reportStepNum() - 1) : timer.reportStepNum();
```

Comment:
I think I get why this happens, now.

To give an example, with 10 1-day timesteps, we name them report steps 0-9. When we want the state at the end of report step n, we want to give it the "sequence number" (n+1), so that the "sequence numbers" are 0-10, with 0 being the initial state and 10 being the end state. To accomplish this, we increment the timer *before* calling the output, so that the output *after* step n will have the number (n+1). As you have discovered this goes wrong because the output facility then tries to access the schedule of the next step.

I think that your solution will prevent crashes, but probably cause the "sequence numbers" written to the output file to be wrong (one less than they should). A full solution must involve opm-output, and while I have not read the code there fully, I assume that the "report_step" passed is also used as sequence number? If so, that must be amended to increment it by one. @joakim-hove or @jokva could probably say something about it?

Finally: a better solution than this one would be to switch the order of the timer increment and output statement in SimulatorBase_impl.hpp. Can you check if that works for you, @blattms?